### ISSUE TITLE
Update generated ESLint config file for ESLint 9

### ISSUE THREAD

**Issue by KnightYoshi** [wxt-dev/wxt#710]
### Feature Request

Update ESLint config to support the ESLint 9 structure.

### What are the alternatives?

I have managed to get ESLint to recognize the WXT globals by importing it as:
```
import globals from 'globals';
import wxtimport from './.wxt/eslintrc-auto-import.json' assert { type: 'json' };

export default [
    {
        languageOptions: {
            globals: {
                ...globals.browser,
                ...wxtimport.globals,
            },
        },
    }
```

### Additional context

ESLint 9 has deprecated the `.eslintrc*` files in favor of eslint.config.js, eslint.config.mjs, and eslint.config.cjs and changed the config structure

https://eslint.org/docs/latest/use/configure/configuration-files

Looked through how it's generated and got lost trying to follow the trail 🤣

**Comment by aklinker1** [comment]
Or, a much simpler alternative, just output a second file file along-side the existing file with the following content:

```ts
// .wxt/eslint-v9-config.js
import globals from 'globals';
import v8Config from './.wxt/eslintrc-auto-import.json' assert { type: 'json' };

export default {
        languageOptions: {
            globals: {
                ...globals.browser,
                ...v8Config.globals,
            },
        },
    }
```

And tell people to extend that file for v9 and up. Eventually, I'd like to just generate one file, but this could be a short-term solution.

**Comment by KnightYoshi** [comment]
Thanks for pointing me in the right direction. I don't mind giving it a shot myself. I ended up in the [getUnimportOptions](https://github.com/wxt-dev/wxt/blob/34ea23030b932da7a66f0942d45025fd37bc236d/packages/wxt/src/core/utils/building/resolve-config.ts#L268C16-L268C34) function trying to find where it happens 🙃 

I've looked at how the _globals_ package does it, that package just has an `index.js` script that exports the `globals.json` file. I think we can do something similar here that way it's not backwards breaking.

I'll see about doing this a little later when I have some free
